dawn/src/dawn/AST/ASTExpr.h

```diff
@@ -566,9 +566,10 @@ class LiteralAccessExpr : public Expr {
 /// @ingroup sir
 class ReductionOverNeighborExpr : public Expr {
 private:
+  enum OperandKind { Rhs = 0, Init };
+
   std::string op_ = "+";
-  std::shared_ptr<Expr> rhs_;
-  std::shared_ptr<Expr> init_;
+  std::array<std::shared_ptr<Expr>, 2> operands_;
```

Comment:
not that I dislike it, I'm just curious why the change?

Reply:
I don't really like it, but it's needed because I implement `getChildren` now and to implement that, I need the array (as it is done for `BinaryOperator`, for example).

Reply:
got it.